llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -3803,12 +3803,7 @@ static unsigned ComputeNumSignBitsImpl(const Value *V,
     case Instruction::SDiv: {
       const APInt *Denominator;
       // sdiv X, C -> adds log(C) sign bits.
-      if (match(U->getOperand(1), m_APInt(Denominator))) {
-
-        // Ignore non-positive denominator.
-        if (!Denominator->isStrictlyPositive())
-          break;
-
+      if (match(U->getOperand(1), m_StrictlyPositive(Denominator))) {
```

Comment:
Aren't you now allowing poison values with m_StrictlyPositive whereas m_APInt didn't allow them?